### PR TITLE
Update AzureDevops plugin

### LINKED ISSUE #6132: [Bug][Schema] cicd_pipelines lacks `started_date`, `queued_date` and cicd_tasks lacks `created_date`, `queued_date`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

When investigating into the CI/CD schema, I found that
- table cicd_pipelines lacks the field `started_date`
- table cicd_tasks lacks the field `created_date`

These two fields are different in CI/CD workflows. For instance, if a GitLab job is queued after it's created, its `created_date` and `started_date` will be different.

image

### What do you expect to happen
In v0.20:
- [x] Add the field `started_date` and `queued_date` to table cicd_pipelines
- [x] Add the field `created_date`and `queued_date` to table cicd_tasks
- [x] Add the field `queued_date` to table cicd_deployment_commits
- [x] Check and fix the value assignment logic of Bamboo's `created_date`, `queued_date` and `started_date`
- [x] Check and fix the value assignment logic of GitLab's `created_date`, `queued_date` and `started_date`
- [x] Update the schema doc and diagram: https://devlake.apache.org/docs/DataModels/DevLakeDomainLayerSchema#cicd_pipelines

In v0.21:
- [x] Check the values of `created_date` and `started_date` in cicd_pipelines
   - [x] GitHub 
   - [x] Jenkins
   - [x] BitBucket
   - [x] Azure
- [x] Check the values of `created_date` and `started_date` in cicd_tasks
   - [x] GitHub 
   - [x] Jenkins
   - [x] BitBucket
   - [x] Azure
- [x] Check the values of `created_date`, `queued_date` and `started_date` in cicd_deployment_commits
   - [x] GitHub 
   - [x] Jenkins
   - [x] BitBucket
   - [ ] Azure

### How to reproduce

Collect a GitLab job that has been queued.

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. Add original_status and original_result fields
2. Add started_date\created_data\queued_data\finished_date fields, update their values and keep the same with other plugins
3. Add new test connection api, parts of #5643
4. Fix a bug when calculating duration(cc @keon94 )

### Does this close any open issues?
Closes #6132 #6139 

### Screenshots
Include any relevant screenshots here.
#### New Test Connection API
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/server/services/remote/plugin/connection_api.go [hunk 1]
```diff
@@ -35,6 +37,7 @@ type TestConnectionResult struct {
 
 func (pa *pluginAPI) TestConnection(input *plugin.ApiResourceInput) (*plugin.ApiResourceOutput, errors.Error) {
 	var result TestConnectionResult
+	fmt.Println("body:", input.Body)
```
[the remark this answers is not in the document]
Comment (d4x1) [reply backend/server/services/remote/plugin/connection_api.go]: It should be removed.

File: backend/python/plugins/azuredevops/azuredevops/streams/builds.py [hunk 2]
```diff
@@ -37,26 +37,26 @@ def convert(self, b: Build, ctx: Context):
...
+        result = devops.CICDResult.RESULT_DEFAULT
         if b.result == Build.BuildResult.Canceled:
-            result = devops.CICDResult.ABORT
+            result = devops.CICDResult.FAILURE
         elif b.result == Build.BuildResult.Failed:
             result = devops.CICDResult.FAILURE
         elif b.result == Build.BuildResult.PartiallySucceeded:
-            result = devops.CICDResult.SUCCESS
-        elif b.result ==  Build.BuildResult.Succeeded:
+            result = devops.CICDResult.FAILURE
```
Comment (abeizn) [inline backend/python/plugins/azuredevops/azuredevops/streams/builds.py:46]: Is this a failure or a success?

Comment (d4x1) [reply backend/python/plugins/azuredevops/azuredevops/streams/builds.py:46]: Honestly speaking, I think it should be considered as "FAILURE" althrougth its name is `PartiallySucceeded`.

Comment (d4x1) [reply backend/python/plugins/azuredevops/azuredevops/streams/builds.py:46]: Comfirmed, `PartiallySucceeded ` can be regared as "FAILURE".

**Review by abeizn — APPROVED** [review]
lgtm

**Pull request merged**